bentoml/_internal/frameworks/onnxmlir.py

```diff
@@ -0,0 +1,396 @@
+
+
+@contextmanager
+def setup_execution_session() -> t.Generator[t.Type[ExecutionSession], None, None]:
+    original_sys_path = deepcopy(sys.path)
+    try:
+        if ONNX_MLIR_RUNTIME:
+            sys.path.append(ONNX_MLIR_RUNTIME)
+        if not TYPE_CHECKING:
+            from PyRuntime import ExecutionSession
+
+            yield ExecutionSession
+    except ImportError:
+        raise MissingDependencyException(warning)
+    finally:
+        sys.path = original_sys_path  # this is safer to do than pop the last element.
+
+
+def get(tag_like: str | Tag) -> bentoml.Model:
+    model = bentoml.models.get(tag_like)
+    if model.info.module not in (MODULE_NAME, __name__):
+        raise NotFound(
+            f"Model {model.tag} was saved with module {model.info.module}, not loading with {MODULE_NAME}."
+        )
+    return model
+
+
+def load_model(
+    bento_model: str | Tag | bentoml.Model, use_default_entry_point: bool = True
+) -> ExecutionSession:
+    """
+    Load ONNXMLir compiled model with the given tag from the local BentoML model store.
+
+    Args:
+        bento_model: Either the tag of the model to get from the store, or a BentoML `~bentoml.Model` instance to load the model from.
+        use_default_entry_point: use the default entry point that is :code:`run_main_graph` or not. Set to :code:`True` by default.
+
+    Returns:
+        :obj:`ExecutionSession`: that can be used to run inference, that is loaded from the model store or BentoML :obj:`~bentoml.Model`.
+
+    Example:
+    .. code-block:: python
+
+        from __future__ import annotations
+
+        from typing import TYPE_CHECKING
+
+        import bentoml
+        import numpy as np
+
+        if TYPE_CHECKING:
+            from bentoml.onnxmlir import ExecutionSession
+
+        session: ExecutionSession = bentoml.onnxmlir.load_model("compiled_mnist:latest")
+        inputs = np.full((1, 1, 28, 28), 1, np.dtype(np.float32))
+        session.run([inputs])
+    """  # noqa
+    if not isinstance(bento_model, bentoml.Model):
+        bento_model = get(bento_model)
+
+    if bento_model.info.module not in (MODULE_NAME, __name__):
+        raise NotFound(
+            f"Model {bento_model.tag} was saved with module {bento_model.info.module}, not loading with {MODULE_NAME}."
+        )
+    use_default_entrypoint: bool = (
+        bento_model.info.options.use_default_entry_point or use_default_entry_point
+    )
+
+    with setup_execution_session() as Session:
+        return Session(
+            shared_lib_path=bento_model.path_of(MODEL_FILENAME),
+            use_default_entry_point=use_default_entrypoint,
+        )
+
+
+def _get_onnxmlir_version() -> str:  # pragma: no cover (trivial)
+    import requests
+
+    version_url = "https://raw.githubusercontent.com/onnx/onnx-mlir/main/VERSION_NUMBER"
+    return requests.get(version_url).text.strip()
+
+
+def save_model(
+    name: str,
+    model_path: Path | PathType,
```

Comment:
This means that we accept both `.so` and `.onnx`. For saving `.so` `trust_compiled_model` has to set to `True`.